src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -129,6 +129,24 @@ private struct Entry {
             }
         }
 
+        public Dictionary(IEnumerable<KeyValuePair<TKey, TValue>> collection):
+            this(collection, null) { }
+
+        public Dictionary(IEnumerable<KeyValuePair<TKey, TValue>> collection, IEqualityComparer<TKey> comparer):
+            this(comparer)
+        {
+            if (collection == null) {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.collection);
+            }
+
+            var coll = collection as ICollection<KeyValuePair<TKey, TValue>>;
+            Initialize(coll?.Count ?? 0);
```

Comment:
This is going to call `Initialize(0)` when `coll == null`. Would it be better to avoid the call to `Initialize` altogether when `coll == null`? That would align with the behavior of not calling `Initialize` from `.ctor(int capacity)` when `capacity` is `0` and from `.ctor(IDictionary<TKey, TValue> dictionary)` when `dictionary` is `null`.

So something like:

```c#
var coll = collection as ICollection<KeyValuePair<TKey, TValue>>;
if (coll != null) {
    Initialize(coll.Count);
}
```